src/raft.rs

```diff
@@ -1281,10 +1397,80 @@ impl<T: Storage> Raft<T> {
                 pr.become_probe();
             }
             ProgressState::Replicate => pr.ins.free_to(m.index),
+            // In the leader's ProgressSet, the flow control of a `Delegated` progress is turned off
+            ProgressState::Delegated => {}
         }
         *maybe_commit = true;
     }
 
+    fn handle_append_response_in_delegate(
```

Comment:
Seems only difference between this and `handle_append_response` is the latter needs to handle leader transferring. That logic is compatible for delegates. So I think this function is unnecessary.